src/pipx/venv.py

```diff
@@ -34,6 +40,16 @@
 
 logger = logging.getLogger(__name__)
 
+# This regex was only introduced into importlib.metadata in Python 3.9.
```

Comment:
You could use `sys.version_info < (3, 9)` instead of the try/except above, then require importlib_metadata for < 3.9 instead of < 3.8. Just a thought.

Reply:
The regex supplied in `importlib.metadata` on >=3.9 is private; `importlib.metadata` uses it to implement additional properties on the `EntryPoint` class so `module` `attr` `extras` can be accessed directly. I’d prefer not to use it since it makes the fallback implementation needlessly complicated.

Reply:
I’ve rewritten the comment to better describe the intention.